fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -847,16 +847,18 @@ def _set_is_root(self) -> None:
         for n, m in self.named_modules():
             # `n != ""` excludes self.
             if n != "" and isinstance(m, FullyShardedDataParallel):
-                assert m._is_root is None
-                m._is_root = False
-                # When root instance doesn't have params, allow children instances
-                # to queue the post_backward hook.
-                #
-                # TODO (Min): we should think if we can have a empty param at the root
-                #             so that root always have a callback on the backward graph.
-                if not self._has_params:
-                    assert m._queue_wait_for_post_backward_closure is None
-                    m._queue_wait_for_post_backward_closure = self._queue_wait_for_post_backward
+                # We relax the assert for non-root instance. A lightning unit test triggers this otherwise.
```

Comment:
not sure if we need to mention lightning here inside of fairscale. eventually this comment will also be unclear what it was relaxed from or why its relaxed

Reply:
I don't want to leave no comment so that it is might be hard to figure out in the future why was this relaxed. Maybe you can suggest something clearer?